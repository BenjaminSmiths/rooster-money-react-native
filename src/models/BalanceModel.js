import {observable} from 'mobx';
import ChildrenStore from '../stores/ChildrenStore'

export default class BalanceModel {
    childUsername: string
    @observable charityPotBalance: number
    @observable goalBalance: number
    @observable savingsBalance: number
    @observable totalBalance: number
    @observable walletBalance: number

    store: ChildrenStore

    constructor(store, {childUsername, goalBalance, savingsBalance, totalBalance, walletBalance}) {
        this.store = store
        this.childUsername = childUsername;
        this.goalBalance = goalBalance;
        this.savingsBalance = savingsBalance;
        this.totalBalance = totalBalance;
        this.walletBalance = walletBalance;
    }

    destroy() {
        this.store.balances.remove(this);
    }

    toJS() {
        return {
            store: this.store,
            childUsername: this.childUsername,
            goalBalance: this.goalBalance,
            savingsBalance: this.savingsBalance,
            totalBalance: this.totalBalance,
            walletBalance: this.walletBalance
        };
    }

    static fromJS(store, object) {
        return new BalanceModel(store, object);
    }
}
