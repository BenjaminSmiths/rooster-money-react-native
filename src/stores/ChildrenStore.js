import { observable, computed, reaction, action } from 'mobx';
import { API_URL, ACCESS_KEY, ACCESS_PASSWORD } from 'react-native-dotenv'

import parentStore, { ParentStore } from './ParentStore'

import BalanceModel from '../models/BalanceModel'

export default class ChildrenStore {
    @observable balances = [];
    @observable goals = [];

    parentStore: ParentStore

    getBalance() {
        reaction(
            () => this.parentStore.isAuthenticated,
            () => {
                return fetch(`${API_URL}/v1/balance`, {
                    method: 'get',
                    headers: new Headers({
                        'Content-Type': 'application/json',
                        'x-access-token': this.parentStore.getToken
                    })
                })
                    .then(response => response.json())
                    .then(balances => {
                        this.balances = balances.map(balance => new BalanceModel(this, balance))
                    })
            }
        );
    }

    static create(parentStore) {
        const childrenStore = new ChildrenStore();
        childrenStore.parentStore = parentStore
        return childrenStore;
    }
}