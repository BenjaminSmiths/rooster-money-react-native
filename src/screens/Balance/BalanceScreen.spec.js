import React from 'react';
import { Provider } from 'mobx-react';
import renderer from 'react-test-renderer';
import BalanceModel from '../../models/BalanceModel'
import ChildrenStore from '../../stores/ChildrenStore'
import parentStore from '../../stores/ParentStore';
import BalanceSceen from './BalanceScreen';


test('Balance Screen Renders Correctly', () => {
    // Given
    const childStore = ChildrenStore.create(parentStore)
    const balanceModel = new BalanceModel(childStore, {
        childUsername: 'test13465',
        goalBalance: 50,
        savingsBalance: 4,
        totalBalance: 54,
        walletBalance: 0,
    })
    const mockNavigation = { state: { params: { balanceModel } } };

    // When
    const tree = renderer.create(
        <Provider>
            <BalanceSceen navigation={mockNavigation}/>
        </Provider>
    ).toJSON();

    // Then
    expect(tree).toMatchSnapshot();
});