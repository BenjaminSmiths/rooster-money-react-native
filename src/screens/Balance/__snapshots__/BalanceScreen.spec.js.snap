// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Balance Screen Renders Correctly 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Array [
      Object {
        "alignItems": "stretch",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/images/background-details.jpg",
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
  <View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "height": 300,
            "justifyContent": "flex-end",
            "marginBottom": 30,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
              "fontSize": 50,
            },
          ]
        }
      >
        €54.00
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
            },
          ]
        }
      >
        ACCOUNT TOTAL
      </Text>
      <Image
        source={
          Object {
            "testUri": "../../../src/images/ChildIcon-Mock.png",
          }
        }
        style={
          Array [
            Object {
              "marginTop": 30,
            },
          ]
        }
      />
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          },
        ]
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#fdc52f",
              "borderColor": "transparent",
              "borderRadius": 1042,
              "borderStyle": "solid",
              "borderWidth": 1,
              "flexBasis": 0,
              "flexGrow": 0,
              "flexShrink": 1,
              "justifyContent": "center",
              "minHeight": 180,
              "minWidth": 180,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              },
              null,
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#fff",
                  "fontSize": 20,
                },
              ]
            }
          >
            €0.00
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#fff",
                },
              ]
            }
          >
            Spend
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#da485f",
              "borderColor": "transparent",
              "borderRadius": 1042,
              "borderStyle": "solid",
              "borderWidth": 1,
              "flexBasis": 0,
              "flexGrow": 0,
              "flexShrink": 1,
              "justifyContent": "center",
              "minHeight": 150,
              "minWidth": 150,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              },
              null,
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#fff",
                  "fontSize": 20,
                },
              ]
            }
          >
            €50.00
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#fff",
                },
              ]
            }
          >
            Goal
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          },
        ]
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#b0b0b0",
              "borderColor": "gray",
              "borderRadius": 1042,
              "borderStyle": "dashed",
              "borderWidth": 4,
              "flexBasis": 0,
              "flexGrow": 0,
              "flexShrink": 1,
              "justifyContent": "center",
              "minHeight": 100,
              "minWidth": 100,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              },
              null,
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#fff",
                  "fontSize": 20,
                },
              ]
            }
          >
            €0.00
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#fff",
                },
              ]
            }
          >
            Give
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#b0b0b0",
              "borderColor": "gray",
              "borderRadius": 1042,
              "borderStyle": "dashed",
              "borderWidth": 4,
              "flexBasis": 0,
              "flexGrow": 0,
              "flexShrink": 1,
              "justifyContent": "center",
              "minHeight": 100,
              "minWidth": 100,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              },
              null,
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#fff",
                  "fontSize": 20,
                },
              ]
            }
          >
            €4.00
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#fff",
                },
              ]
            }
          >
            Save
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
